packages/core/src/result-transformers.ts

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Record, { Dict } from './record'
+import Result from './result'
+import EagerResult from './result-eager'
+import ResultSummary from './result-summary'
+import { newError } from './error'
+
+async function createEagerResultFromResult<Entries extends Dict> (result: Result): Promise<EagerResult<Entries>> {
+  const { summary, records } = await result
+  const keys = await result.keys()
+  return new EagerResult<Entries>(keys, records, summary)
+}
+
+type ResultTransformer<T> = (result: Result) => Promise<T>
+/**
+ * Protocol for transforming {@link Result}.
+ *
+ * @typedef {function<T>(result:Result):Promise<T>} ResultTransformer
+ * @interface
+ * @experimental
+ *
+ * @see {@link resultTransformers} for provided implementations.
+ * @see {@link Driver#executeQuery} for usage.
+ * @see https://github.com/neo4j/neo4j-javascript-driver/discussions/1052
+ */
+/**
+ * Defines the object which holds the common {@link ResultTransformer} used with {@link Driver#executeQuery}.
+ *
+ * @experimental
+ * @see https://github.com/neo4j/neo4j-javascript-driver/discussions/1052
+ */
+class ResultTransformers {
+  /**
+   * Creates a {@link ResultTransformer} which transforms {@link Result} to {@link EagerResult}
+   * by consuming the whole stream.
+   *
+   * This is the default implementation used in {@link Driver#executeQuery}
+   *
+   * @example
+   * // This:
+   * const { keys, records, summary } = await driver.executeQuery('CREATE (p:Person{ name: $name }) RETURN p', { name: 'Person1'}, {
+   *   resultTransformer: neo4j.resultTransformers.eagerResultTransformer()
+   * })
+   * // equivalent to:
+   * const { keys, records, summary } = await driver.executeQuery('CREATE (p:Person{ name: $name }) RETURN p', { name: 'Person1'})
+   *
+   *
+   * @experimental
+   * @returns {ResultTransformer<EagerResult<Entries>>} The result transformer
+   * @see https://github.com/neo4j/neo4j-javascript-driver/discussions/1052
+   */
+  eagerResultTransformer<Entries extends Dict = Dict>(): ResultTransformer<EagerResult<Entries>> {
+    return createEagerResultFromResult
+  }
+
+  /**
+   * Creates a {@link ResultTransformer} which maps the {@link Record} in the result and collects it
+   * along with the {@link ResultSummary} and {@link Result#keys}.
+   *
+   * NOTE: The config object requires map or/and collect to be valid.
+   *
+   * @example
+   * // Mapping the records
```

Comment:
it would be interesting to explain here what the default collect does

Reply:
The default collector will return:

```
interface {
   records: R[]
   summary: ResultSummary
   keys: string[]
}```

So, it's quite similar to the eager result. However, the records are already mapped to `R` type defined in the `map` function.